lib/os/heap.c

```diff
@@ -18,17 +18,17 @@ static void *chunk_mem(struct z_heap *h, chunkid_t c)
 	return ret;
 }
 
-static void free_list_remove_bidx(struct z_heap *h, chunkid_t c, int bidx)
+static void free_list_remove_bidx(struct z_heap *h, chunkid_t c, unsigned int bidx)
 {
 	struct z_heap_bucket *b = &h->buckets[bidx];
 
 	CHECK(!chunk_used(h, c));
 	CHECK(b->next != 0);
-	CHECK(h->avail_buckets & (1 << bidx));
+	CHECK(h->avail_buckets & (1U << bidx));
```

Comment:
Ideally we should have a BIT32 macro instead of BIT but introducing it would have a lot of problems due to the several different uses.
However once decided how it is better to build the masks for avail_buckets we should be uniform (and I agree that the two cases shown above are not).